ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -471,7 +422,7 @@ private TopicDescription getTopicDescription(final Admin admin, final String sou
     final ListOffsetsResult listOffsetsResult = admin.listOffsets(
         topicPartitions,
         new ListOffsetsOptions(
-            isolationLevel
+            IsolationLevel.READ_UNCOMMITTED
```

Comment:
Why read uncommitted rather than the code from before?

Reply:
I hard-coded the stream pull query app to be "at least once" (because it is pure side-effect, there is no point in EOS), which means that we will only need to get the end offsets in "read uncommitted" mode.

Reply:
Ok, I added a comment to clarify this.